packages/babel-plugin/src/shared/stylex-position-try.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow strict
+ */
+
+import type { InjectableStyle, StyleXOptions } from './common-types';
+
+import createHash from './hash';
+import expandShorthands from './preprocess-rules/index';
+import generateLtr from './physical-rtl/generate-ltr';
+import generateRtl from './physical-rtl/generate-rtl';
+import transformValue from './utils/transform-value';
+import dashify from './utils/dashify';
+import {
+  objFromEntries,
+  objEntries,
+  objMapKeys,
+  objMap,
+  Pipe,
+} from './utils/object-utils';
+import { defaultOptions } from './utils/default-options';
+
+// Similar to `stylex.keyframes` but for position-try-fallbacks
+// Takes an object of positioning properties and returns a string after hashing it
+// The generated string must be prefixed with -- for anchor positioning
+export default function styleXPositionTry(
+  styles: { +[string]: string | number },
+  options: StyleXOptions = defaultOptions,
+): [string, InjectableStyle] {
+  const { classNamePrefix = 'x' } = options;
+  const expandedObject = Pipe.create(styles)
+    .pipe((styles) => expandPositionShorthands(styles, options))
+    .pipe((x) => objMapKeys(x, dashify))
+    .pipe((x) => objMap(x, (value, key) => transformValue(key, value, options)))
+    .done();
+
+  const ltrStyles = objMap(expandedObject, (value, key) =>
+    generateLtr([key, value]),
+  );
+  const rtlStyles = objMap(
+    expandedObject,
+    (value, key) => generateRtl([key, value]) ?? value,
+  );
+
+  const ltrString = constructPositionTryObj(ltrStyles);
+  const rtlString = constructPositionTryObj(rtlStyles);
+
+  const positionTryName = '--' + classNamePrefix + createHash(ltrString);
+
+  const ltr = `@position-try ${positionTryName} {${ltrString}}`;
+  const rtl =
```

Comment:
Let's not add this RTL code to new APIs

Reply:
AFAIK, the RTL code is still used at for Meta codebases. Other changes are needed before this code can be removed. I'll start working on removing that code as it should be safe, even at Meta.

Reply:
It might be used for existing APIs, but we don't need it for this one. This API won't be used at Meta for a while, and I don't think we need to propagate the issues with RTL to new APIs in general.